notebooks/name-search-utility-demo.ipynb

```diff
@@ -0,0 +1,331 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1. Introduction\n",
+    "This notebook demonstrates a super simple, but potentially super useful function to find elements of the nwm network based on their GNIS name. \n",
+    "\n",
+    "The scripts used to develop the GNIS names table used in this script have not been maintained and we are working on updating them, so, for the meantime, we must warn that any results from this tool are to be considered only partially reproducible.  \n",
+    "\n",
+    "## 2. First steps\n",
+    "Import basic libraries, set a root path, get connected to the repository if operating on colab, and set the path search directory."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sys\n",
+    "import os\n",
+    "import pathlib\n",
+    "import json\n",
+    "import xarray as xr\n",
+    "import pandas as pd\n",
+    "\n",
+    "try:\n",
+    "    import google.colab\n",
+    "\n",
+    "    ENV_IS_CL = True\n",
+    "    root = pathlib.Path(\"/content/t-route\").resolve()\n",
+    "    subprocess.run([\"git\", \"clone\", \"https://github.com/NOAA-OWP/t-route.git\"])\n",
+    "except:\n",
+    "    root = pathlib.Path(\"..\").resolve()\n",
+    "sys.path.append(os.path.join(root, \"src\", \"python_framework_v02\"))\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3. Import named segments\n",
+    "### 3a. Import the raw .json file\n",
+    "This file is a dictionary organized as:\n",
+    "```\n",
+    "{\"Name of River as Key Value\":[list,of,nwm,link,ids,with,that,name],\n",
+    " \"Another Name River\":[xxxx,yyyy,etc.]}\n",
+    "```\n",
+    "That structure minimizes duplication. However, it should be noted that it also means that every Sand Creek is listed under the same `\"Sand Creek\"` key and that care must be taken in interpreting the results."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "json_names_path = os.path.join(\n",
+    "    root, r\"test\", \"input\", \"json\", r\"nwm_reaches_conus_v21_wgnis_name.json\",\n",
+    ")\n",
+    "\n",
+    "with open(json_names_path, \"r\") as json_file:\n",
+    "    names = json.load(json_file)\n",
+    "    if names:\n",
+    "        print(f\"success!! -- imported {len(names)} GNIS names from the table.\")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 3b. Convert the names into a dataframe keyed on segment id"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ids_wnames = {}\n",
+    "for name, ids in names.items():\n",
+    "    for id in ids:\n",
```

Comment:
`id` is a very important builtin function in Python. Please use another identifier here.

Reply:
Ok. Good catch.